Makefile

```diff
@@ -0,0 +1,245 @@
+
+##############################################################################
+# Variables used for various build targets.
+##############################################################################
+
+# Ensure the Go version in .go_version is installed and used.
+GOROOT?=$(shell ./scripts/make/run_with_go_ver go env GOROOT)
+GO:=$(GOROOT)/bin/go
+export PATH:=$(GOROOT)/bin:$(PATH)
+
+# By default we run tests with verbose output. This may be overridden, e.g.
+# scripts may set GOTESTFLAGS=-json to format test output for processing.
+GOTESTFLAGS?=-v
+
+GOOSBUILD:=./build/$(shell $(GO) env GOOS)
+APPROVALS=$(GOOSBUILD)/approvals
+GENPACKAGE=$(GOOSBUILD)/genpackage
+GOIMPORTS=$(GOOSBUILD)/goimports
+GOLICENSER=$(GOOSBUILD)/go-licenser
+GOLINT=$(GOOSBUILD)/golint
+MAGE=$(GOOSBUILD)/mage
+REVIEWDOG=$(GOOSBUILD)/reviewdog
+STATICCHECK=$(GOOSBUILD)/staticcheck
+ELASTICPACKAGE=$(GOOSBUILD)/elastic-package
+
+PYTHON_ENV?=.
+PYTHON_BIN:=$(PYTHON_ENV)/build/ve/$(shell $(GO) env GOOS)/bin
+PYTHON=$(PYTHON_BIN)/python
+
+CLOUDBEAT_VERSION=$(shell grep defaultBeatVersion cmd/version.go | cut -d'=' -f2 | tr -d '" ')
+
+# Create a local config.mk file to override configuration,
+# e.g. for setting "GOLINT_UPSTREAM".
+-include config.mk
+
+##############################################################################
+# Rules for building and unit-testing cloudbeat.
+##############################################################################
+
+.DEFAULT_GOAL := cloudbeat
+
+.PHONY: cloudbeat
+cloudbeat:
+	@$(GO) build -o $@
+
+.PHONY: test
+test:
+	$(GO) test $(GOTESTFLAGS) ./...
+
+.PHONY:
+clean: $(MAGE)
+	@$(MAGE) clean
+
+##############################################################################
+# Checks/tests.
+##############################################################################
+
+.PHONY: check-full
+# check-full: update check golint staticcheck check-docker-compose
```

Comment:
why?

Reply:
The other commands are used for pr builds,update is the only one relevant for main branches I can remove that line - i suggest to clean it up in a batch.

Reply:
I'm not sure I understood your answer, maybe I wasn't clear with my comment. There are two `check-full` commands, one is commented out which I assume really checking the entire flow (`update`, `check`, `golint`, `staticcheck`, `check-docker`,`compose`) and the other one only runs `update`. Is this by design? from what I see it's being used in the intake file so I'm curious of why you made it check less stuff, does the other checks fail?

Reply:
Exactly, some failed others applied formatting to the code, and were intended for pr branches.So i left the minimum required for the snapshot build.

Reply:
okay if we have failing test we should address them, although this indeed not relevant to this PR we should document this failing test and add them to our CI

Reply:
These checks aren't mandatory and not necessarily have to be implemented exactly how they are done here. if we decide to do it, they need to go in our pr checks since it's mostly styling & formating.
The reason i disabled them, is because they will otherwise fmt/lint every file - which would've resulted in massive conflicts with main.

the checks are:

1. mage check-fmt & mage check-headers
2. honnef.co/go/utils/cmd/staticcheck
3. golint
4. docker integration test